File: doc/src_OpenGL/DepotGL.cpp
// DepotGL.cpp: implementation of the DepotGL class.
//
//////////////////////////////////////////////////////////////////////
#include "Stdafx.h"
#include "OpenGL.h"

#include "Inkludy.h"

#ifdef _DEBUG
#undef THIS_FILE
static char THIS_FILE[]=__FILE__;
#define new DEBUG_NEW
#endif

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

DepotGL::DepotGL()
{
	list = 60;
}

DepotGL::~DepotGL()
{

}

void DepotGL::draw(WorldObject* object)
{
	glCallList(list);
}

void DepotGL::init()
{
	glNewList(list,GL_COMPILE);
		#include "depot"
	glEndList();
}

File: app/src/main/java/robot/object/opengl/DepotGL.java
package robot.object.opengl;

import robot.TextureLoader;
import robot.object.Depot;
import robot.object.WorldObject;
import robot.object.opengl.ObjectGL;

import javax.media.opengl.GL;
import javax.media.opengl.GL;
import java.nio.FloatBuffer;

public class DepotGL extends ObjectGL {

    Depot depot;

    public DepotGL() {
        list = 60;
    }

    public void draw(GL gl, WorldObject object) {
        gl.glPushMatrix();
        gl.glBindTexture(GL.GL_TEXTURE_2D, TextureLoader.array[TextureLoader.START]);
        gl.glTranslatef(0f,0f,0.5f);
        gl.glCallList(list);
        gl.glPopMatrix();

    }

    public void init(GL gl) {
        gl.glPushMatrix();
        gl.glNewList(list,GL.GL_COMPILE);
        CubeGL.createCube(gl,0.8f,true);
        gl.glEndList();
        gl.glPopMatrix();
    }
}